docs/DeveloperGuide.md

```diff
@@ -1,34 +1,670 @@
+* [Appendix B: Guidelines on Manual Testing](#appendix-b-guidelines-on-manual-testing)
+* [Appendix C: Effort](#appendix-c-effort)
 
+## 1.0 Introduction
 
-## Product scope
-### Target user profile
+**LifEasier** helps Computer Engineering (CEG) students to manage their classes, social events, deadlines and school notes through the Command Line Interface (CLI). **LifEasier** is developed by a group of CEG students for their CS2101/CS2113T mods.
 
-{Describe the target user profile}
+This developer guide documents the design, architecture and instructions for testing, as a reference for developers who will be maintaining or expanding **LifEasier** in the future. 
 
-### Value proposition
+## 2.0 Setting Up
 
-{Describe the value proposition: what problem does it solve?}
+The following section will provide information on how to set up **LifEasier** for development.
 
-## User Stories
+### 2.1 Prerequisites
+
+1. `JDK 11`
+2. Intellij IDE
+
+### 2.2 Setting Up the Project
+
+1. Open _IntelliJ_ (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+
+1. Set up the correct JDK version for Gradle.
+Click `Configure` > `Project Defaults` > `Project Structure`
+Click `New...` and find the directory of the JDK.
+
+1. Click Import Project.
+
+1. Locate the `build.gradle` file and select it. Click `OK`.
+
+1. Click `Open as Project`.
+
+1. Click `OK` to accept the default settings.
+
+### 2.3 Verifying Setup
+
+1. Run the `seedu.LifEasier.Main` and try a few commands.
+
+1. Run the tests to ensure they all pass. This can be done by executing the command `gradlew build` in _IntelliJ’s_ terminal.
+
+
+## 3.0 Design
+
+The following sections will go into detail on the design and architecture of **LifEasier**.
+
+### 3.1 Architecture
+
+This section elaborates on the high-level architecture of the **LifEasier** application. It provides a brief introduction to each component, and how these components interact with one another. Figure 3.1-1 provides an overview of the architecture.
+
+![Figure 3.1-1](images/DeveloperGuide/Figure%203.1-1.png) 
+_Figure 3.1-1 Architecture diagram for LifEasier_
+
+LifEasier is comprised of 7 components, which are listed below together with their functions:
+
+1. `LifEasier`: The main class of the application.
+1. `Ui`: Displays messages to the user, and takes in the user’s commands
+1. `Parser`: Understands the user’s commands, and creates the necessary Command objects.
+1. `Command`: Carries out the user’s command.
+1. `Model`: Holds the user's task and note data in memory.
+1. `Storage`: Reads and writes data from the hard disk.
+
+Each of these components are expanded on in more detail in their respective sections.
+
+Figure 3.1-2 illustrates the Sequence diagram for how each class interacts with one another through an example  “addDeadline” command.
+
+![Figure 3.1-2](images/DeveloperGuide/Figure 3.1-2.png)  
+_Figure 3.1-2. Sequence diagram of “addDeadline”._
```

Comment:
When you call the self-activation bar the text is covering the arrow, maybe it would be clearer if you move the text?

<img width="849" alt="Screenshot 2020-10-29 at 12 22 24 PM" src="https://user-images.githubusercontent.com/57402349/97525375-9c90bf80-19e1-11eb-9cc6-af357947b9f2.png">

Reply:
![image](https://user-images.githubusercontent.com/60388723/97528049-ce0c8980-19e7-11eb-9680-02ed3a422ae6.png)
Shouldn't the arrow be pointing to the start of the activation bar?